packages/fast-tooling-react/src/form/form/badge-control.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { BadgeControlProps } from "./badge.props";
+import InfoBadge from "./badge.info";
+import LockedBadge from "./badge.locked";
+import WarningBadge from "./badge.warning";
+import { BadgeType } from "./form-item.props";
+
+export default class BadgeControl extends React.Component<BadgeControlProps, {}> {
+    public render(): React.ReactNode {
+        switch (this.props.type) {
+            case BadgeType.info:
+                return (
+                    <InfoBadge
+                        className={this.props.className}
+                        description={this.props.description}
+                    />
+                );
+            case BadgeType.warning:
+                return (
+                    <WarningBadge
+                        className={this.props.className}
+                        description={this.props.description}
+                    />
+                );
+            case BadgeType.locked:
+                return (
+                    <LockedBadge
+                        className={this.props.className}
+                        description={this.props.description}
+                    />
```

Comment:
These three badges are very similar and expect the same data - seems a bit heavy to me to create 3 badges and this `BadgeControl` when you can have just one component. Is there a reason we need four?

Reply:
That's true, I can consolidate them